copilot/src/App/App.js

```diff
@@ -1,28 +1,129 @@
 import AgateDecorator from '@enact/agate/AgateDecorator';
-import kind from '@enact/core/kind';
+import openSocket from 'socket.io-client';
 import qs from 'query-string';
+import {Cell, Row} from '@enact/ui/Layout';
+import kind from '@enact/core/kind';
 import React from 'react';
-
+import Button from '@enact/agate/Button';
 import css from './App.less';
+import Popup from '@enact/agate/Popup';
 
 const args = qs.parse(typeof window !== 'undefined' ? window.location.search : '');
 
-const App = kind({
+const AppBase = kind({
 	name: 'App',
 
 	styles: {
 		css,
-		className: 'app'
+		className: 'app enact-fit'
 	},
 
-	render: (props) => {
-		const {url} = args;
+	render: ({adContent, showAd, url, popupOpen, setScreen, togglePopup, ...rest}) => {
 		return (
-			<div {...props}>
-				<iframe src={url} />
-			</div>
+			<Row {...rest}>
+				<Popup
```

Comment:
I'm not sure how I feel about the semantics of the `Popup` and the `Button` being considered part of the `Row`.

What do you think of this being a `Panel` or just a simple `div` that has the `Row`, `Button`, and `Popup` as its children?

Reply:
Yeah i made it a simple `div` because there are no other Panels for now. Don't need the overhead.